src/cycax/cycad/assembly.py

```diff
@@ -36,18 +36,16 @@ def render(self):
             # FIXME: Should just be: `part.render()` the part render should sort out its own stuff.
             name = part["part_no"]
 
-            STLname = "{cwd}/{name}/{name}.stl".format(cwd=self._base_path, name=name)
-            if not os.path.exists(STLname):
-                SCADname = "{cwd}/{name}/{name}.scad".format(cwd=self._base_path, name=name)
-                if not os.path.exists(SCADname):
-                    logging.info("Creating a SCAD file of the pieces of the object.")
+            stl_name = "{cwd}/{name}/{name}.stl".format(cwd=self._base_path, name=name)
+            if not os.path.exists(stl_name):
```

Comment:
We should always create the files. Best is to call render on the part.

If we wish to do caching (something for later) or only compile/create STL when needed then we need to create a hash of new scad and old scad. If hash is different create the STL file. 
You could take that a step further and only create files if JSON changed.
If JSON of parts changed also recreate Assembly STL.